shared/pkg/analyzer/merge.go

```diff
@@ -59,33 +62,50 @@ func (m *MergedResults) Merge(other *Results, format string) *MergedResults {
 		log.Errorf("Failed to decode results: %v", err)
 		return m
 	}
-	if m.SrcMetaData == nil {
-		m.SrcMetaData = bom.Metadata
-	} else {
-		component := mergeCDXComponent(*m.SrcMetaData.Component, *bom.Metadata.Component, true)
-		m.SrcMetaData.Component = &component
-	}
+
+	// merge bom.Metadata.Component if it exists
+	m.mergeMainComponent(bom)
+
 	if other.AppInfo.SourceHash != "" {
 		m.addSourceHash(other.AppInfo.SourceHash)
 	}
-	if bom.Components == nil {
-		log.Debugf("Decoded bom doesn't contain any components")
-		return m
-	}
-	otherComponentsByKey := toComponentByKey(bom.Components)
-	for key, otherComponent := range otherComponentsByKey {
-		if mergedComponent, ok := m.MergedComponentByKey[key]; !ok {
-			log.Debugf("Adding new component results from %v. key=%v", other.AnalyzerInfo, key)
-			m.MergedComponentByKey[key] = newMergedComponent(otherComponent, other.AnalyzerInfo)
-		} else {
-			log.Debugf("Adding existing component results from %v. key=%v", other.AnalyzerInfo, key)
-			m.MergedComponentByKey[key] = handleComponentWithExistingKey(mergedComponent, otherComponent, other.AnalyzerInfo)
+
+	if bom.Components != nil {
+		otherComponentsByKey := toComponentByKey(bom.Components)
+		for key, otherComponent := range otherComponentsByKey {
+			if mergedComponent, ok := m.MergedComponentByKey[key]; !ok {
+				log.Debugf("Adding new component results from %v. key=%v", other.AnalyzerInfo, key)
+				m.MergedComponentByKey[key] = newMergedComponent(otherComponent, other.AnalyzerInfo)
+			} else {
+				log.Debugf("Adding existing component results from %v. key=%v", other.AnalyzerInfo, key)
+				m.MergedComponentByKey[key] = handleComponentWithExistingKey(mergedComponent, otherComponent, other.AnalyzerInfo)
+			}
 		}
 	}
 
+	if bom.Dependencies != nil {
+		// TODO merge dependencies after normalize
+		m.Dependencies = m.normalizeDependencies(bom.Dependencies)
+	}
+
 	return m
 }
 
+func (m *MergedResults) mergeMainComponent(bom *cdx.BOM) {
+	// Keep track of all SBOM refs given to the main component so that we
+	// can normalize them later.
+	if bom.Metadata != nil && bom.Metadata.Component != nil && bom.Metadata.Component.BOMRef != "" {
+		m.SrcMetaDataBomRefs = append(m.SrcMetaDataBomRefs, bom.Metadata.Component.BOMRef)
+	}
+
+	if m.SrcMetaData == nil {
+		m.SrcMetaData = bom.Metadata
+	} else {
+		component := mergeCDXComponent(*m.SrcMetaData.Component, *bom.Metadata.Component, true)
```

Comment:
should we validate that m.SrcMetaData.Component != nil and bom.Metadata.Component != nil?